ui/js/graphs.js

```diff
@@ -243,10 +243,23 @@ perf.controller('GraphsCtrl', [
         series.flotSeries.lines.show = false;
       });
 
-      $("#overview-plot").bind("plotselected", function (event, ranges) {
-        deselectDataPoint();
+      $("#overview-plot").bind("plotunselected",function() {
+        $scope.zoom = {};
+        $scope.selectedDataPoint = null;
         hideTooltip();
+        updateDocument();
+        // refetch and re-render all graph data
+        $q.all($scope.seriesList.map(getSeriesData)).then(function() {
+          plotGraph();
+        });
```

Comment:
You don't need to refetch the data here, I don't think, just re-render it. So remove the q.all() promise, and inline the plotGraph() -- probably something like:

    updateDocument()
    plotGraph(); // re-render all graph data